shared/editor/nodes/Mention.ts

```diff
@@ -0,0 +1,192 @@
+import Token from "markdown-it/lib/token";
+import { InputRule } from "prosemirror-inputrules";
+import { NodeSpec, Node as ProsemirrorNode, NodeType } from "prosemirror-model";
+import { EditorState, TextSelection, Plugin } from "prosemirror-state";
+import { MarkdownSerializerState } from "../lib/markdown/serializer";
+import { run } from "../plugins/BlockMenuTrigger";
+import isInCode from "../queries/isInCode";
+import mentionRule from "../rules/mention";
+import { Dispatch, EventType } from "../types";
+import Node from "./Node";
+
+const OPEN_REGEX = /(?:^|\s)@([a-zA-Z+-]+)?$/;
+const CLOSE_REGEX = /(?:^|\s)@(([0-9a-zA-Z_+-]*\s+)|(\s+[0-9a-zA-Z_+-]+)|[^0-9a-zA-Z_+-]+)$/;
```

Comment:
These need to be changed to support `i18n` :see_no_evil:

Reply:
I hope it's safe to use https://github.com/tc39/proposal-regexp-unicode-property-escapes :crossed_fingers:

Reply:
I wonder if we can just use `{Letter}` here, or more specifically `\p{Letter}\d` any letter in any language and digits.

Reply:
Are digits needed for name?

Reply:
By the way, it's recommended to include _combining marks_ as well.

>“character” really means “Unicode code point”. \p{L} matches a single code point in the category “letter”. If your input string is à encoded as U+0061 U+0300, it matches a without the accent. If the input is à encoded as U+00E0, it matches à with the accent. The reason is that both the code points U+0061 (a) and U+00E0 (à) are in the category “letter”, while U+0300 is in the category “mark”.

:point_up: from https://www.regular-expressions.info/unicode.html#prop

 So `\p{L}{M}` should be good.

Reply:
I probably would just because folks signing in with different OIDC providers often end up with a username-like string in the name field. It doesn't hurt